docs/fsharp/tour.md

```diff
@@ -1,4 +1,158 @@
+
+[!code-fsharp[Tuples](../../samples/snippets/fsharp/tour.fs#L114-L121)]
+
+Note that struct tuples and "regular" tuples are distinct types.  You cannot "fool" the compiler into thinking one is the other.  Instead, you must explicitly transfer the values from one tuple type to the other if you wish to work with both types of tuples.
+
+## Arrays, Lists, and Sequences
+
+Arrays, Lists, and Sequences are three primary collection types in the F# core library.
+
+[Arrays](language-reference/arrays.md) are fixed-size, *mutable* collections of elements of the same type.  They are also all loaded into memory, thus they support fast random access of elements.
+
+[!code-fsharp[Arrays](../../samples/snippets/fsharp/tour.fs#L246-L276)]
+
+[Lists](language-reference/lists.md) are ordered, immutable collections of elements of the same type.  Under the covers, they are singly-linked lists, which makes them lightweight and good for enumeration, but a poor choice for random access and concatenation if they're large.
+
+[!code-fsharp[Lists](../../samples/snippets/fsharp/tour.fs#L129-L161)]
+
+[Sequences](language-reference/sequences.md) are a logical series of elements, all of the same type.  These are a more general type that Lists and Arrays, capable of being your "view" into any logical series of types.  They also stand out because they are *lazy*, which means that elements are computed only when they are needed.
+
+[!code-fsharp[Sequences](../../samples/snippets/fsharp/tour.fs#L289-L315)]
+
+## Recursive Functions
+
+Processing collections or sequences of elements is typically done with [recursion](language-reference/functions/index.md#recursive-functions) in F#.  Although F# has support for loops and imperative programming, recursion is preferred because it is easier to guarantee correctness.
+
+>![NOTE]
+This same makes use of the pattern matching via the `match` expression.  This fundamental construct is covered later in this article.
+
+[!code-fsharp[RecursiveFunctions](../../samples/snippets/fsharp/tour.fs#L323-L349)]
+
+F# also has full support for "tail call optimization", which for the uninitiated is a way to optimize recursive calls so that they are just as fast as a loop construct.  This is fairly unique, as most languages require implementing things like Stack Trampolines to avoid overflowing the program Stack with recursive calls.
+
+## Record and Discriminated Union Types
+
+Record and Union types are two fundamental data types used in F# code, and are generally the best way to represent data in an F# program.  Although this makes them similar to classes in other languages, one of their primary differences is that they have structural equality semantics.  This means that they are "natively" comparable and equality is straightforward - just check if one is equal to the other.
```

Comment:
Ah-Ha! Regarding my previous comment on Tuples. I was thinking of Records. It might make sense to reference this section where Tuples are introduced.